src/components/Modals/NewCourse/CourseConflictModal.vue

```diff
@@ -0,0 +1,163 @@
+      <span v-if="numTotalConflicts <= 1">
+        This course can fulfill these requirements. Select one:
+      </span>
+      <div v-else>
+        <span>Please fix the following</span
+        ><span class="courseConflict--bold">{{ ` ${numTotalConflicts} ` }}</span
+        ><span>conflicts</span>
+      </div>
+    </div>
+
+    <div v-for="index in numTotalConflicts" :key="index" class="courseConflict-conflict">
+      <div v-if="numTotalConflicts > 1">{{ `${index}. Choose only one requirement:` }}</div>
+      <single-conflict-editor :conflictNumber="index" @conflict-changed="handleChangedConflict" />
+      <div v-if="index === 1" class="courseConflict-warning">
+        <span>*Requirements with</span>
+        <img class="warning-icon" src="@/assets/images/warning.svg" alt="warning icon" />
+        <span>are broad so check carefully before selecting them</span>
+      </div>
+    </div>
+  </TeleportModal>
+</template>
+
+<script lang="ts">
+import { PropType, defineComponent } from 'vue';
+import TeleportModal from '@/components/Modals/TeleportModal.vue';
+import SingleConflictEditor from '@/components/Modals/NewCourse/SingleConflictEditor.vue';
+
+export default defineComponent({
+  components: { TeleportModal, SingleConflictEditor },
+  emits: {
+    'close-course-modal': () => true,
+    'add-course': (course: CornellCourseRosterCourse, choice: FirestoreCourseOptInOptOutChoices) =>
+      typeof course === 'object' && typeof choice === 'object',
+  },
+  props: {
+    selectedCourse: { type: Object as PropType<FirestoreSemesterCourse>, required: true },
+  },
+  data() {
+    // TODO @willespencer remove hardcoded initial list of reqs
+    return {
+      selectedReqsPerConflict: [
+        ['req1', 'req2', 'selectableReq1'],
+        ['req1', 'req2', 'selectableReq1'],
+      ],
+      numConflictsUnresolved: 2,
+    };
+  },
+  computed: {
+    conflictsResolved(): boolean {
+      return this.numConflictsUnresolved === 0;
+    },
+    rightButtonText(): string {
+      return this.conflictsResolved ? 'Done' : 'Save';
+    },
+    numTotalConflicts(): number {
+      return this.selectedReqsPerConflict.length;
+    },
+    errorText(): string {
+      if (!this.conflictsResolved) {
+        return 'Conflict: Please only choose one requirement';
+      }
+      return '';
+    },
+  },
+  methods: {
+    closeCurrentModal() {
+      this.$emit('close-course-modal');
+    },
+    addItem() {
+      this.addCourse();
+    },
+    addCourse() {
+      // TODO @willespencer handle what happens when conflicts are saved or resolved
+      this.closeCurrentModal();
+    },
+    handleChangedConflict(selectedReqs: string[], index: number) {
+      this.selectedReqsPerConflict[index - 1] = selectedReqs;
+      this.updateNumUnresolvedConflicts();
+    },
+    // conflicts exist for each list of reqs in selectedReqsPerConflict if number of reqs selected != 1
+    updateNumUnresolvedConflicts() {
```

Comment:
This function may need to be changed. It's not necessarily the case that a constraint violation means only one requirement can be chosen. For example, if (A,B,C) is in the requirements list because of constraints between A-B and A-C, then unchecking A but keeping B and C would suffice. The steps should be something like

1. generate requirements lists from `getConstraintViolationsForSingleCourse(currentGraph)`
2. user updates edges, we create a new temporary subgraph from those choices
3. check constraint violations with `getConstraintViolationsForSingleCourse(tempSubgraph)`

Reply:
@einc maybe the wording in the designs should be changed to reflect this, idk

Reply:
Discussed offline with @einc - since we cannot come up with a concrete example right now, we will push this off and thoroughly test after the rest has been implemented!